src/RPCClient.ts

```diff
@@ -91,7 +91,7 @@ class RPCClient<M extends ClientManifest> {
     logger.info(`Created ${this.name}`);
     return rpcClient;
   }
-
+  protected idGen?: IdGenFunction;
```

Comment:
This will not be optional. It will always be `protected idGen: IdGen;`. Use defaulted function of `() => null` on the creation.

Don't specify defaults in constructor or in the property initialiser here.

Because it is parameterised, it should only be defaulted in **one** place. That is in the creation under CD and CDSS. Or constructor under SS patterns.